ragna/deploy/_ui/components/metadata_filters_builder.py

```diff
@@ -0,0 +1,307 @@
+from datetime import datetime
+
+import panel as pn
+import param
+
+from ragna.core._metadata_filter import MetadataFilter
+
+METADATA_FILTERS = {
+    "document_name": {"type": str},
+    "size": {"type": int},
+    "extension": {"type": str},
+    "section": {"type": str},
+    # "creation_date": {"type": datetime},
+}
+
+
+FILTERS_PER_TYPE = {
+    str: ["==", "!=", "in", "not in"],
+    int: ["==", "!=", ">", "<", ">=", "<=", "in", "not in"],
+    # datetime: ["==", "!=", ">", "<", ">=", "<="],
+}
+
+PLACEHOLDERS_PER_METADATA_KEY = {
+    "document_name": "",
+    "size": "",
+    "extension": "",
+    "section": "",
+    # "creation_date": "YYYY-mm-dd HH:MM:SS",
+}
+
+
+class FilterRow(pn.viewable.Viewer):
+    key = param.Selector(
+        objects=[
+            "",  # empty one for the first row
+            *METADATA_FILTERS.keys(),
+        ],
+        default="",
+    )
+    operator = param.Selector(
+        objects=[
+            "",  # empty one for the first row
+            # Filled later based on the key
+            # "raw"
```

Comment:
What does "raw" mean here?

Reply:
Just a remainder of something I reworked afterwards. I remove it